eZ/Publish/Core/Repository/Mapper/ContentDomainMapper.php

```diff
@@ -119,6 +127,19 @@ public function buildContentDomainObject(
 
         $versionInfo = $this->buildVersionInfoDomainObject($spiContent->versionInfo, $prioritizedLanguages);
 
+        $contentInfo = $versionInfo->getContentInfo();
+        $mainLocation = $contentInfo->getMainLocation();
+        $contentLocations = $this->locationHandler->loadLocationsByContent($contentInfo->id);
```

Comment:
This operation is unfortunately too heavy for the mapper. We need here new method for Persistence handler, so you can call:
```suggestion
        $contentLocationsCount = $this->locationHandler->countContentItemLocations($contentInfo->id);
```
It needs to be defined on Location Persistence Handler interface, handler implementation, handler cache implementation and have dedicated Gateway method.

Reply:
@alongosz apparently there was already a gateway method `countLocationsByContentId` made - https://github.com/ezsystems/ezplatform-kernel/blob/1.3/eZ/Publish/Core/Persistence/Legacy/Content/Location/Gateway.php#L333. I've just implemented it into handler - please, see https://github.com/ezsystems/ezplatform-kernel/pull/364/commits/ae63e9aa0c644d0e2e88e02c9ead4bf9cc216cba